docs/resources/azure_generic_resource.md

```diff
@@ -1,27 +1,80 @@
+---
+title: About the azure_generic_resource Resource
+---
 
-The following table describes the options that can be passed to the resource
+# azure_generic_resource
 
-| Name | Description |
-|------|-------------|
-| group_name | Name of the resource group in which to find resources | 
-| name | Name of the specific resource to look for |
-| type | Type of resources to look for |
-| apiversion | When using the `name` option to look for a specific resource an API Version can be set. If not set the latest version available for that type of resource will be used |
+Use the `azure_generic_resource` Inspec audit resource to test any valid Azure Resource. This is very useful if you need to test something that we do not yet have a specific Inspec resource for.
+
+## Syntax
+
+```ruby
+describe azure_generic_resource(group_name: 'MyResourceGroup', name: 'MyResource') do
+  its('attribute') { should eq 'value' }
+end
+```
+
+where
+
+* `MyResourceGroup` is the name of the resource group that contains the Azure Resource to be validated
+* `MyResource` is the name of the resource that needs to be checked
+* `attribute` - This generic resource dynamically creates the attributes on the fly based on the type of resource that has been targetted.
+* `value` is the expected output from the chosen attribute
+
+The options that can be passed to the resource are as follows.
+
+| Name        | Description                                                                                                         | Required | Example                           |
+|-------------|---------------------------------------------------------------------------------------------------------------------|----------|-----------------------------------|
+| group_name: | Azure Resource Group to be tested                                                                                   | yes      | MyResourceGroup                   |
+| name:       | Name of the Azure resource to test                                                                                  | no       | MyVM                              |
+| type: | Azure Resource Type to look for | no | Microsoft.Compute/virtualMachines |
+| apiversion: | API Version to use when interrogating the resource. If not set then the latest version for the resoure type is used | no       | 2017-10-9                         |
 
-As an alternative the following environment variables can be set instead of passing the options to the control:
+These options can also be set using the environment variables:
 
  - `AZURE_RESOURCE_GROUP_NAME`
  - `AZURE_RESOURCE_NAME`
  - `AZURE_RESOURCE_TYPE`
  - `AZURE_RESOURCE_API_VERSION`
 
-Note that if options are set on the control and in enviroment variables, the environment variables take perference.
+When the options have been set as well as the environment variables, the environment variables take priority.
```

Comment:
It was done in this way so that thing scould easily be tested without having to modify a file.
I have worked on some CIS controls for Azure, which @jamesc asked for, that can use this concept to quickly test things.